trieste/acquisition/function.py

```diff
@@ -170,30 +181,39 @@ def __repr__(self) -> str:
         return "ExpectedImprovement()"
 
     def prepare_acquisition_function(
-        self, dataset: Dataset, model: ProbabilisticModel
+        self,
+        model: ProbabilisticModel,
+        dataset: Optional[Dataset] = None,
```

Comment:
Can overloads be used here to avoid that? That, and also occasions where dataset isn't even used, but is declared anyway. Just a wild thought, I don't really know if that's possible